py-polars/polars/internals/dataframe/frame.py

```diff
@@ -6809,7 +6804,12 @@ def iter_rows(
         if buffer_size:
             for offset in range(0, self.height, buffer_size):
                 zerocopy_slice = self.slice(offset, buffer_size)
-                if named and _PYARROW_AVAILABLE:
+                if (
```

Comment:
Can we set a variable of this check outside of the loop? This will be evaluated more than once right?

Reply:
Quite right; lifted the whole check out.